src/components/DataDictionary/dataDictionary.tsx

```diff
@@ -1,16 +1,38 @@
-import React from "react";
-import { BaseComponentProps } from "../types";
+import React, { useMemo } from "react";
 import { Entities } from "./components/Entities/entities";
+import { EntitiesLayout as DefaultEntitiesLayout } from "./components/Layout/components/EntitiesLayout/entitiesLayout";
+import { OutlineLayout as DefaultOutlineLayout } from "./components/Layout/components/OutlineLayout/outlineLayout";
+import { TitleLayout as DefaultTitleLayout } from "./components/Layout/components/TitleLayout/titleLayout";
+import { Outline as DefaultOutline } from "./components/Outline/outline";
+import { buildClassesOutline } from "./components/Outline/utils";
+import { Title as DefaultTitle } from "./components/Title/title";
 import { View } from "./dataDictionary.styles";
 import { useDataDictionary } from "./hooks/UseDataDictionary/hook";
+import { useLayoutSpacing } from "./hooks/UseLayoutSpacing/hook";
+import { DataDictionaryProps } from "./types";
 
 export const DataDictionary = ({
   className,
-}: BaseComponentProps): JSX.Element => {
+  EntitiesLayout = DefaultEntitiesLayout,
+  Outline = DefaultOutline,
+  OutlineLayout = DefaultOutlineLayout,
+  Title = DefaultTitle,
+  TitleLayout = DefaultTitleLayout,
+}: DataDictionaryProps): JSX.Element => {
```

Comment:
`DataDictionary` component props `DataDictionaryProps` now includes optional layouts and also optional title, outline. If not specified the defaults are used.